transformations/style_paraphraser/paraphraser_helpers/style_paraphraser.py

```diff
@@ -0,0 +1,394 @@
+                )
+            )
+            logits = pred__["logits"]
+            past = pred__["past_key_values"]
+            next_token_logits = logits[:, -1, :] / (
+                temperature if temperature > 0 else 1.0
+            )
+            filtered_logits = _top_k_top_p_filtering(
+                next_token_logits, top_k=top_k, top_p=top_p
+            )
+            do_greedy = temperature == 0 and top_k in [0, 1] and top_p == 0.0
+            next_token = (
+                torch.argmax(filtered_logits, dim=-1).unsqueeze(-1)
+                if do_greedy
+                else torch.multinomial(
+                    F.softmax(filtered_logits, dim=-1), num_samples=1
+                )
+            )
+            generated = torch.cat((generated, next_token), dim=1)
+            segments = torch.cat((segments, segments[:, -1:]), dim=1)
+
+            for batch_elem in range(batch_size):
+                if next_token[batch_elem].item() == eos_token_id:
+                    eos_emitted[batch_elem] = True
+
+            if length is None and all(eos_emitted):
+                break
+    return generated
+
+
+def _score_fn(x, length_normalize):
+    if length_normalize:
+        return x["score"] / len(x["sequence"])
+    else:
+        return x["score"]
+
+
+class Instance(object):
+    """
+    Instance of a sentence to generate a paraphrase for.
+
+    Args:
+        args: Standard, default arguments to use.
+        config : configurations specified.
+        instance_dict : dictionary containing the sentence to paraphrase.
+
+    """
+
+    def __init__(self, args, config, instance_dict):
+        self.dict = instance_dict
+        self.args = args
+        self.config = config
+        self.truncated = False
+        self.sent1_tokens = np.array(
+            instance_dict["sent1_tokens"], dtype=np.int32
+        )
+        self.sent2_tokens = np.array(
+            instance_dict["sent2_tokens"], dtype=np.int32
+        )
+        self.init_context_size = config["max_prefix_length"] + 1
+
+    def preprocess(self, tokenizer):
+        self.truncate()
+        self.build_sentence(tokenizer)
+        self.build_label(tokenizer)
+        self.build_segment(tokenizer)
+
+    def truncate(self):
+        config = self.config
+        max_prefix_length = config["max_prefix_length"]
+        max_suffix_length = config["max_suffix_length"]
+        if len(self.sent1_tokens) > max_prefix_length:
+            self.truncated = True
+            self.sent1_tokens = self.sent1_tokens[:max_prefix_length]
+        if len(self.sent2_tokens) > max_suffix_length:
+            self.truncated = True
+            self.sent2_tokens = self.sent2_tokens[:max_suffix_length]
+
+    def build_sentence(self, tokenizer):
+        tokens_to_pad = self.config["max_prefix_length"] - len(
+            self.sent1_tokens
+        )
+        self.sent_prefix = np.pad(
+            self.sent1_tokens,
+            (tokens_to_pad, 0),
```

Comment:
Just a bit curious, why is the first sentence padded on the left? Is the model trained with the same format of input so that it should be preprocessed like this?

Reply:
Hi @Nickeilf, I'm the original author of the style transfer paper. Yes we padded to the left during training to keep a constant input length, so we continued to do this during inference.